pkg/collector/corechecks/systemd/dbus_conn.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+// +build systemd
+
+package systemd
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/coreos/go-systemd/dbus"
+	godbus "github.com/godbus/dbus"
+)
+
+// NewSystemdConnection establishes a private, direct connection to systemd.
+// This can be used for communicating with systemd without a dbus daemon.
+// Callers should call Close() when done with the connection.
+// Note: method borrowed from `go-systemd/dbus` to provide custom path for systemd private socket
```

Comment:
What do you mean here ? Can it be imported and used directly then?
If not, any licensing issues?

Reply:
1) The origin method is like that: https://github.com/coreos/go-systemd/blob/master/dbus/dbus.go#L143-L150

And this method is not public: https://github.com/coreos/go-systemd/blob/master/dbus/dbus.go#L204-L222
that's why I reproduced it here.

2) I added a link to the original source/repo. The original License is Apache 2.0 https://github.com/coreos/go-systemd/blob/master/LICENSE

Reply:
Also added `This product includes software developed at CoreOS, Inc (http://www.coreos.com/).` to comply with https://github.com/coreos/go-systemd/blob/master/LICENSE